karamel-core/src/main/java/se/kth/karamel/backend/dag/RecipeSerialization.java

```diff
@@ -48,6 +49,7 @@ public void prepareToExecute(RunRecipeTask task) throws InterruptedException {
           + " at the moment because parallelism is limited. Available parallelism permits: "
           + parallelism.availablePermits() + "/" + maxParallelism + " - we wait until a permit becomes available."
           + " Current claims: " + printableClaims());
+      task.blocked();
```

Comment:
If the recipe parallelism is set, it will first `tryAcquire` the lock. If it succeeds then it moves on and executes the recipe. If it fails to `tryAcquire` it means there are not enough free locks so set the state to BLOCKED and then wait until `acquire`. This BLOCKED state is only used in the UI, it's not used anywhere else